packages/database/src/migrations/20210129022440-AddFacilityTypeOverlayForFiji-modifies-data.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+import { codeToId, insertObject } from '../utilities';
+import { generateId } from '../utilities/generateId';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const MAIN_OVERLAY_ID = 'Fiji_Operational_Facilities';
+
+const LINKED_OVERLAY_ID = 'Fiji_Operational_Facility_Type';
+
+const MAIN_OVERLAY = {
+  id: MAIN_OVERLAY_ID,
+  name: 'Operational facilities',
+  userGroup: '',
+  dataElementCode: 'BCD1',
+  linkedMeasures: `{${LINKED_OVERLAY_ID}}`,
+  isDataRegional: true,
+  measureBuilderConfig: {
+    entityAggregation: {
+      dataSourceEntityType: 'facility',
+    },
+  },
+  measureBuilder: 'valueForOrgGroup',
+  presentationOptions: {
+    values: [
+      {
+        icon: 'circle',
+        name: 'Open',
+        value: ['Fully Operational', 'Operational but closed this week'],
+      },
+      {
+        icon: 'x',
+        name: 'Temporarily closed',
+        value: 'Temporarily Closed',
+      },
+      {
+        icon: 'triangle',
+        name: 'Permanently closed',
+        value: 'Permanently Closed',
+      },
+      {
+        icon: 'empty',
+        name: 'No data',
+        value: 'null',
+      },
+    ],
+    displayType: 'icon',
+    customColors: 'RoyalBlue,RoyalBlue,OrangeRed,OrangeRed',
+    measureLevel: 'Facility',
+  },
+  countryCodes: '{FJ}',
+  projectCodes: '{explore}',
+};
+
+const LINKED_OVERLAY = {
+  id: LINKED_OVERLAY_ID,
+  name: 'Facility types',
+  dataElementCode: 'facilityTypeCode',
+  userGroup: 'Public',
+  measureBuilderConfig: {
+    facilityTypeCodeMetadataKey: 'type',
+  },
+  measureBuilder: 'valueForOrgGroup',
+  presentationOptions: {
+    values: [
```

Comment:
Was it necessary to specify these values? I would have expected using `displayedValueKey` would be enough, and make it less "stuck" to the config and more able to respond to whatever the facility types might get changed to in future?

Reply:
As I understand (I might be wrong), even with `displayedValueKey` defined, we still have to specify the values for the overlay. Looking at (`updateLegendFromDisplayedValueKey` in `measureData.js`), it seems to me is what it does is it will try to override the labels/names specified in `values` with the defined `displayedValueKey`, given that the `values` still has to be specified with the correct `value`...?

Reply:
Aah cool, thanks for clarifying!